src/commcare_cloud/ansible/roles/pgbouncer/tasks/main.yml

```diff
@@ -44,18 +60,90 @@
     reload: yes
   with_dict: "{{ pgbouncer_kernel_settings }}"
 
-- name: pgbouncer init script
-  template: src=pgbouncer.init.j2 dest=/etc/init.d/pgbouncer
-  notify: Restart pgbouncer
+- name: pgbouncer enumerate current processes
+  set_fact:
+    pgbouncer_current_processes: "{{ pgbouncer_current_processes | default([]) | union([item.value.name | regex_search('pgbouncer-multiprocess\\@(\\d+)\\.service', '\\1') | first | int]) }}"
+  loop: "{{ lookup('dict', ansible_facts.services) }}"
+  when: item.key | regex_search('pgbouncer-multiprocess@(\d+)\.service')
+  no_log: true
+  tags:
+    - pgbouncer
+    - configure
+
+- name: pgbouncer stop and disable excess processes
+  become: yes
+  systemd:
+    name: "pgbouncer-multiprocess@{{ item }}"
+    state: stopped
+    enabled: no
+  with_sequence: start="{{ pgbouncer_processes + 1 }}" end="{{ pgbouncer_current_processes | max }}"
+  when: pgbouncer_current_processes | max > pgbouncer_processes
```

Comment:
=this task and the next 2 could be put into a block with a single when which helps to show that they are all related. Alternately the could be extracted to a separate file and only included if necessary.

```
block:
  - name: pgbouncer stop and disable excess processes
  - name: pgbouncer remove conf for excess processes
  - name: pgbouncer remove unix socket directory for excess processes
when: pgbouncer_current_processes | max > pgbouncer_processes
```

Reply:
Added them in a block. There's only 3 so unless we need to call them from elsewhere I think it's ok to keep them in the main file.